Add Crust to dStorage section (#962)
near/docs: docs/concepts/decentralized-storage.md

:::tip
Please note that using Arweave's network will require purchasing artokens to pay for storage. Learn more at [arweave.org](https://www.arweave.org/).
:::

## Crust

[Crust](https://crust.network) provides a Web3.0 decentralized storage network for the Metaverse. It is designed to realize core values of decentralization, privacy and assurance. Crust supports multiple storage-layer protocols such as IPFS, and exposes instant accessible on-chain storage functions to users. Crustʼs technical stack is also capable of supporting data manipulating and computing.

> The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) procotol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust based on Polkadot ecosystem, but also supports the majority contract platforms, including Near/Solana/Ethereum/Elrond/... with its cross-chain solution.

Reviewer: ```suggestion
The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) protocol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust is based on the Polkadot ecosystem, but also supports the majority of contract platforms, including Near/Solana/Ethereum/Elrond with its cross-chain solution.
```

> The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) protocol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust is based on the Polkadot ecosystem, but also supports the majority of contract platforms, including Near/Solana/Ethereum/Elrond/... with its cross-chain solution.

Reviewer: ```suggestion
The Crust protocol is 100% compatible with [IPFS](https://ipfs.io) protocol, it matches people who have hard drive space to spare with those users that need to store data or host content. Crust is based on the Polkadot ecosystem, but also supports the majority of contract platforms, including NEAR/Solana/Ethereum/Elrond/etc. with its cross-chain solution.
```

:::info
To learn more about Crust, check its [Decentralized Storage Market](https://wiki.crust.network/docs/en/DSM) and [Guaranteed Proof of Stake](https://wiki.crust.network/docs/en/GPoS)
Also, you can start with [Build-101](https://wiki.crust.network/docs/en/build101).
:::

### Example of storing file with Crust and Near

### 1. Upload file to IPFS

> > If you want to learn how to upload **FOLERS/FILES** into IPFS, please refer this [STEP](https://wiki.crust.network/docs/en/buildFileStoringWithGWDemo#1-upload-files-to-ipfs-gateway)

Reviewer: ```suggestion
> If you want to learn how to upload **FOLDERS/FILES** into IPFS, please refer to this [STEP](https://wiki.crust.network/docs/en/buildFileStoringWithGWDemo#1-upload-files-to-ipfs-gateway)
```

> First, you need to put your file into IPFS, there's 2 ways to upload file to IPFS: local IPFS node or remote IPFS gateway:

Reviewer: ```suggestion
First, you need to put your file into IPFS. There are 2 ways to upload files to IPFS: local IPFS node or remote IPFS gateway:
```

- With local IPFS node

```typescript
import { create } from 'ipfs-http-client'

async function addFile(ipfs: IPFS.IPFS, fileContent: any) {
    // 1. Create IPFS instant
    const ipfs = create({
        url: 'http://localhost:5001'
    });

    // 2. Add file to ipfs
    const { cid } = await ipfs.add(fileContent);

    // 3. Get file status from ipfs
    const fileStat = await ipfs.files.stat("/ipfs/" + cid.path);

    return {
        cid: cid.path,
        size: fileStat.cumulativeSiz
    };
}
```

- With [IPFS W3Authed Gateway](https://docs.ipfs.io/concepts/ipfs-gateway/#authenticated-gateways)

> You can find more `ipfsW3GW` endpoints on [LINK](https://github.com/crustio/ipfsscan/blob/main/lib/constans.ts#L29).

> > You can also find more `authHeader` web3 supports on [LINK](https://github.com/RoyTimes/crust-workshop/tree/master/src), the following exmaple just takes Near as example.

Reviewer: ```suggestion
> You can also find more `authHeader` web3 supports on [LINK](https://github.com/RoyTimes/crust-workshop/tree/master/src), the following sample just takes Near as an example.
```
